src/main/java/edu/ie3/datamodel/models/input/thermal/DomesticHotWaterStorageInput.java

```diff
@@ -0,0 +1,180 @@
+   * @param storageVolumeLvl Available storage volume
+   * @param inletTemp Temperature of the inlet
+   * @param returnTemp Temperature of the outlet
+   * @param c Specific heat capacity of the storage medium
+   * @param pThermalMax Maximum thermal power of the storage
+   */
+  public DomesticHotWaterStorageInput(
+      UUID uuid,
+      String id,
+      ThermalBusInput bus,
+      ComparableQuantity<Volume> storageVolumeLvl,
+      ComparableQuantity<Temperature> inletTemp,
+      ComparableQuantity<Temperature> returnTemp,
+      ComparableQuantity<SpecificHeatCapacity> c,
+      ComparableQuantity<Power> pThermalMax) {
+    super(uuid, id, bus, storageVolumeLvl, inletTemp, returnTemp, c, pThermalMax);
+  }
+
+  @Override
+  public DomesticHotWaterStorageInputCopyBuilder copy() {
+    return new DomesticHotWaterStorageInputCopyBuilder(this);
+  }
+
+  @Override
+  public String toString() {
+    return "DomesticHotWaterStorageInput{"
+        + "uuid="
+        + getUuid()
+        + ", id="
+        + getId()
+        + ", operator="
+        + getOperator().getUuid()
+        + ", operationTime="
+        + getOperationTime()
+        + ", bus="
+        + getThermalBus().getUuid()
+        + ", storageVolumeLvl="
+        + getStorageVolumeLvl()
+        + ", inletTemp="
+        + getInletTemp()
+        + ", returnTemp="
+        + getReturnTemp()
+        + ", c="
+        + getC()
+        + ", pThermalMax="
+        + getpThermalMax()
+        + '}';
+  }
+
+  public static class DomesticHotWaterStorageInputCopyBuilder
+      extends CylindricalStorageInputCopyBuilder {
+
+    public DomesticHotWaterStorageInputCopyBuilder(DomesticHotWaterStorageInput entity) {
+      super(entity);
+    }
+
+    @Override
+    public DomesticHotWaterStorageInputCopyBuilder storageVolumeLvl(
+        ComparableQuantity<Volume> storageVolumeLvl) {
+      this.storageVolumeLvl = storageVolumeLvl;
+      return this;
+    }
+
+    @Override
+    public DomesticHotWaterStorageInputCopyBuilder inletTemp(
+        ComparableQuantity<Temperature> inletTemp) {
+      this.inletTemp = inletTemp;
+      return this;
+    }
+
+    @Override
+    public DomesticHotWaterStorageInputCopyBuilder returnTemp(
+        ComparableQuantity<Temperature> returnTemp) {
+      this.returnTemp = returnTemp;
+      return this;
+    }
+
+    @Override
+    public DomesticHotWaterStorageInputCopyBuilder c(ComparableQuantity<SpecificHeatCapacity> c) {
+      this.c = c;
+      return this;
+    }
+
+    @Override
+    public DomesticHotWaterStorageInputCopyBuilder pThermalMax(
+        ComparableQuantity<Power> pThermalMax) {
+      this.pThermalMax = pThermalMax;
+      return this;
+    }
+
+    @Override
+    public DomesticHotWaterStorageInputCopyBuilder scale(Double factor) {
+      storageVolumeLvl(this.storageVolumeLvl.multiply(factor));
+      pThermalMax(this.pThermalMax.multiply(factor));
+      return this;
+    }
```

Comment:
I think you can remove these methods, since they copy the behavior of the superior class.

Reply:
Oh yes, thanks!